cylc/flow/scheduler.py

```diff
@@ -1807,7 +1807,11 @@ async def _shutdown(self, reason: BaseException) -> None:
         sys.stdout.flush()
         sys.stderr.flush()
 
-        if self.contact_data and self.task_job_mgr:
```

Comment:
Related to the previous, because `contact_data` is set a little earlier, this test needed bodging a little as we could get into the situation where the workflow failed to startup, but did configure its contact file triggering remote-tidy to run on shutdown which caused the database to be created when it shouldn't have been.

Test changed to only run when the database is present, which is a better test, as remote-tidy needs database info to function.